platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/StartupStateUtils.java

```diff
@@ -0,0 +1,586 @@
+     * @param swirldName               the name of the swirld
+     * @param currentSoftwareVersion   the current software version
+     * @param emergencyRecoveryManager the emergency recovery manager
+     * @return a reserved signed state (wrapped state will be null if no state could be loaded)
+     * @throws SignedStateLoadingException if there was a problem parsing states on disk and we are not configured to
+     *                                     delete malformed states
+     */
+    @NonNull
+    static ReservedSignedState loadStateFile(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final RecycleBin recycleBin,
+            @NonNull final NodeId selfId,
+            @NonNull final String mainClassName,
+            @NonNull final String swirldName,
+            @NonNull final SoftwareVersion currentSoftwareVersion,
+            @NonNull final EmergencyRecoveryManager emergencyRecoveryManager)
+            throws SignedStateLoadingException {
+
+        final StateConfig stateConfig = platformContext.getConfiguration().getConfigData(StateConfig.class);
+        final String actualMainClassName = stateConfig.getMainClassName(mainClassName);
+
+        final List<SavedStateInfo> savedStateFiles = getSavedStateFiles(actualMainClassName, selfId, swirldName);
+        logStatesFound(savedStateFiles);
+
+        if (savedStateFiles.isEmpty()) {
+            // No states were found on disk.
+            return createNullReservation();
+        }
+
+        final boolean emergencyStateRequired = emergencyRecoveryManager.isEmergencyStateRequired();
+
+        final ReservedSignedState state;
+        if (emergencyStateRequired) {
+            state = loadEmergencyState(
+                    platformContext, recycleBin, currentSoftwareVersion, savedStateFiles, emergencyRecoveryManager);
+        } else {
+            state = loadLatestState(platformContext, recycleBin, currentSoftwareVersion, savedStateFiles);
+        }
+
+        return state;
+    }
+
+    /**
+     * Create a copy of the initial signed state. There are currently data structures that become immutable after being
+     * hashed, and we need to make a copy to force it to become mutable again.
+     *
+     * @param platformContext    the platform's context
+     * @param initialSignedState the initial signed state
+     * @return a copy of the initial signed state
+     */
+    public static @NonNull ReservedSignedState copyInitialSignedState(
+            @NonNull final PlatformContext platformContext, @NonNull final SignedState initialSignedState) {
+        Objects.requireNonNull(platformContext);
+        Objects.requireNonNull(initialSignedState);
+
+        final State stateCopy = initialSignedState.getState().copy();
+        final SignedState signedStateCopy =
+                new SignedState(platformContext, stateCopy, "Browser create new copy of initial state", false);
```

Comment:
Should this message be mentioning the browser? I'm guessing this message was just copied from previous code (also applies to the next message being printed)

Reply:
fixed